src/test/java/org/gridsuite/securityanalysis/server/SecurityAnalysisControllerTest.java

```diff
@@ -93,15 +95,22 @@ public void setUp() {
 
         // action service mocking
         given(actionsService.getContingencyList(CONTINGENCY_LIST_NAME, NETWORK_UUID))
-                .willReturn(Mono.just(SecurityAnalysisFactoryMock.CONTINGENCIES));
+                .willReturn(Flux.fromIterable(SecurityAnalysisFactoryMock.CONTINGENCIES));
         given(actionsService.getContingencyList(CONTINGENCY_LIST2_NAME, NETWORK_UUID))
-                .willReturn(Mono.just(SecurityAnalysisFactoryMock.CONTINGENCIES));
+                .willReturn(Flux.fromIterable(SecurityAnalysisFactoryMock.CONTINGENCIES));
+        given(actionsService.getContingencyList(CONTINGENCY_LIST_ERROR_NAME, NETWORK_UUID))
+                .willReturn(Flux.error(new DataBufferLimitException("Exceeded limit on max bytes to buffer : 262144")));
```

Comment:
use RuntimeException for this exemple, it's more understandable. The DataBufferLimitException won't happen anymore since you used flux instead of mono

Reply:
Maybe return a Flux that correctly sends a few first items, and then signals an error ?

Reply:
Replaced by :
```
given(actionsService.getContingencyList(CONTINGENCY_LIST_ERROR_NAME, NETWORK_UUID))
        .willReturn(Flux.fromIterable(SecurityAnalysisFactoryMock.CONTINGENCIES).thenMany(Flux.error(new RuntimeException("Error message test"))));

```